### PR TITLE
Feature/view mrl results

### PR DESCRIPTION
**Pull request by WernerVdM97**
Summary of this change:
Added a grid showing pass/fail for each mrl chemical result based on mrl chemical requirements.. 

CHANGELOG entry:

### Added
- Menu migration linked to mrl results grid.
- Gemfile changes to support icon columns in grids defined via code.
- Interactor function to calculate pass/fail given mrl requirement id and mrl result id.
- Accompanying repo functions.
### Fixed
- Previous code: qa standards and mrl requirements bugs

### CODE REVIEW

File: lib/quality/repositories/mrl_chemicals_requirement_repo.rb [hunk 1]
```diff
@@ -17,6 +17,10 @@ def chemicals_with_max_eu_lvl
       DB[:mrl_chemicals_requirements].join(:chemicals, id: :chemical_id).select_map(%i[chemical_id chemical_name eu_max_level]).uniq
     end
 
+    def chemical_ids_and_max_lvls_for(mrl_requirement_id)
```
Comment (jcsjcs) [inline lib/quality/repositories/mrl_chemicals_requirement_repo.rb:20]: Rather use the full word levels instead of lvls.

**Review by jcsjcs — CHANGES_REQUESTED** [review]

File: lib/quality/interactors/mrl_result_interactor.rb [hunk 2]
```diff
@@ -107,8 +107,96 @@ def capture_mrl_result(id, params)
...
+      end
+    end
+
+    def row_defs_for_mrl_results_grid(column_ids, mrl_result_ids) # rubocop:disable Metrics/AbcSize
+      mrl_req_repo = MasterfilesApp::MrlRequirementRepo.new
+
+      rows = []
+
+      mrl_result_ids.each do |mrl_result_id|
+        cultivar_id, cultivar_name = repo.cultivar_id_and_name_for(mrl_result_id)
```
[the remark this answers is not in the document]
Comment (WernerVdM97) [reply lib/quality/interactors/mrl_result_interactor.rb]: This makes a lot of sense yes. Please see this solution https://github.com/NoSoft-SA/nspack/pull/370/commits/0839a70a0aea0d8a189b261daa5444959071b597

File: lib/quality/interactors/mrl_result_interactor.rb [hunk 3]
```diff
@@ -136,29 +136,25 @@ def col_defs_for_mrl_results_grid(mrl_requirement_ids_and_column_names)
...
+                season: mrl_result[:season],
+                puc: mrl_result[:puc],
+                farm: mrl_result[:farm],
+                orchard: mrl_result[:orchard] }
 
         column_ids.each do |column_id|
-          tm_type, tm_id = mrl_req_repo.target_market_type_and_id_for(column_id)
-          mrl_requirement_id = mrl_req_repo.mrl_req_id_for(tm_type, tm_id, cultivar_id, where: { season_id: repo.get(:mrl_results, :season_id, mrl_result_id) })
-          result = mrl_passed_for?(mrl_result_id, mrl_requirement_id) unless mrl_requirement_id.nil?
+          tm_type, tm_id = requirements_repo.target_market_type_and_id_for(column_id)
```
Comment (jcsjcs) [inline lib/quality/interactors/mrl_result_interactor.rb:155]: This query does not need to be performed in the loop. It is being run for every column of every row.

Rather load a hash outside the loop - like: `col_types = { column_id => { tm_type: 'abc', tm_id: 123 } }`
Then inside the loop you just reference the hash: `col_types[column_id][:tm_type]` etc.

**Review by jcsjcs — CHANGES_REQUESTED** [review]

File: lib/quality/interactors/mrl_result_interactor.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcsjcs) [reply lib/quality/interactors/mrl_result_interactor.rb]: Yes, that works much better, I have noticed some further optimisations you can do , though - see new comments.

**Pull request merged**